db/silkworm/etl/file_provider.hpp

```diff
@@ -0,0 +1,45 @@
+/*
+   Copyright 2020 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+#ifndef ETL_SILKWORM_FILE_PROVIDER_H
+#define ETL_SILKWORM_FILE_PROVIDER_H
+
+#include <fstream>
+#include <memory>
+#include <optional>
+#include <silkworm/etl/buffer.hpp>
+
+namespace silkworm::etl {
+
+union head_t {
+    uint32_t lengths[2];
+    uint8_t bytes[8];
+};
+
+// FileProvider saves buffers to disk and reads from them
+class FileProvider {
+  public:
+    FileProvider(std::string& working_path, int id);        // Sort and write buffer to file
```

Comment:
1. Pass working_path by **const** reference since you're not changing the argument
2. Constructor's comment is misleading: the constructor only opens a file, it doesn't sort nor writes anything.

Reply:
Right